runtime/lua/vim/lsp/protocol.lua

```diff
@@ -738,14 +738,16 @@ function protocol.make_client_capabilities()
       completion = {
         dynamicRegistration = false,
         completionItem = {
-          -- Until we can actually expand snippet, move cursor and allow for true snippet experience,
-          -- this should be disabled out of the box.
-          -- However, users can turn this back on if they have a snippet plugin.
-          snippetSupport = false,
+          snippetSupport = true,
           commitCharactersSupport = false,
           preselectSupport = false,
           deprecatedSupport = false,
           documentationFormat = { constants.MarkupKind.Markdown, constants.MarkupKind.PlainText },
+          resolveSupport = {
+            properties = {
+              'additionalTextEdits',
```

Comment:
nvim-lsp-compl also includes `'detail'` and `'documentation'` here, but I think we shouldn't since we don't actually resolve these properties when missing in the unresolved completion items.

Any objection @mfussenegger?

Reply:
No objection. 

The idea behind including them was that servers don't have to provide the documentation if it's never displayed - reducing payloads. But if they're displayed would depend on `completeopt`

Reply:
Hmmm I see. It might help with performance then...

The thing is that by setting them here servers will likely never set those properties since they expect them to be resolved, but we never actually resolve them. So I'm not sure...

Reply:
If it's not necessary for the MVP, ignore them -- we can always add them later.